huaweicloud/services/dds/resource_huaweicloud_dds_audit_log_policy.go

```diff
@@ -240,8 +295,19 @@ func resourceDdsAuditLogPolicyDelete(_ context.Context, d *schema.ResourceData,
 		return diag.Errorf("error deleting DDS audit log policy: %s", err)
 	}
 
-	// This is a workaround to avoid db master-slave synchronization
-	time.Sleep(2 * time.Second) // lintignore:R018
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"pending"},
+		Target:       []string{"success"},
+		Refresh:      ddsInstanceAuditLogPolicyRefreshFunc(deleteAuditLogPolicyClient, d.Id()),
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		Delay:        2 * time.Second,
+		PollInterval: 2 * time.Second,
+	}
+
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error waiting for instance (%s) audit log policy to be set  successful: %s", d.Id(), err)
```

Comment:
```suggestion
		return diag.Errorf("error waiting for instance (%s) audit log policy to be deleted successful: %s", d.Id(), err)
```
Are these two statuses also `pending` and `success` when deleted?

Reply:
done, the status of pending and success is custom, use these two status to judge whether the job is completed